Library/Homebrew/test/software_spec_spec.rb

```diff
@@ -218,7 +218,14 @@
     end
   end
 
-  %w[root_url prefix cellar rebuild].each do |method|
+  specify "#cellar" do
```

Comment:
@SeekingMeaning You are changing the externally facing API here. I think the old test needs to stay untouched.

Reply:
I have an alternative implementation for the BottleSpecification class, which introduces a new method:
```
  def cellar_or_default_cellar
    return collector[Utils::Bottles.tag][:cellar] if collector.key?(Utils::Bottles.tag)

    @default_cellar
  end
```

Each time `cellar` is needed that method can be called.

Reply:
Or you could do

```rb
  def cellar(val = nil)
    return collector.dig(Utils::Bottles.tag, :cellar) || default_cellar if val.nil?

    @default_cellar = val
  end
```

Reply:
https://docs.ruby-lang.org/en/2.6.0/Array.html#method-i-dig

> `dig(idx, ...) → object`
> 
> Extracts the nested value specified by the sequence of idx objects by calling dig at each step, returning nil if any intermediate step is nil.
> 
> ```rb
> a = [[1, [2, 3]]]
> 
> a.dig(0, 1, 1)                    #=> 3
> a.dig(1, 2, 3)                    #=> nil
> a.dig(0, 0, 0)                    #=> TypeError: Integer does not have #dig method
> [42, {foo: :bar}].dig(1, :foo)    #=> :bar
> ```

Reply:
Agreed about keeping external test the same. I think ideally, to demonstrate this PR even better, all test changes that don't rely on the new functionality would be moved into a new PR to be merged before this one.

Reply:
I overwrote the one commit made by @SeekingMeaning which reverts the test change, so that we validate that we do not break any legacy tests.

I think I also just realised one thing: `def cellar(val = nil)` is both a getter and a setter, based on the value of the parameter. That's a weird patter I never encountered, and I think I understand better my confusion around all this part of the code.